### PR TITLE
Books dashboard

### PR DESCRIPTION
**Pull request by eledwinn**
## Summary
1. Dashboard & books list
2. Input component

## Screenshot
image

## Trello Card
https://trello.com/c/ZpxLhdGg/15-wbooks

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,11 @@
...
 ## System ##
 .DS_Store
+
+logs/*
+!.gitkeep
+app/lib/
+node_modules/
+tmp
+.DS_Store
+.idea
```
Comment (FrankIglesias) [inline .gitignore]: add EOL

File: app/scss/book-item.scss [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: space-between;
+}
+
+.book-item {
+  display: flex;
+  flex-direction: column;
+  width: 128px;
+  margin: 0 30px 30px 30px;
```
Comment (FrankIglesias) [inline app/scss/book-item.scss]: margin: 0 30px 30px;

File: app/scss/book-item.scss [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+.book-item-image {
+  width: 128px;
```
Comment (FrankIglesias) [inline app/scss/book-item.scss:15]: don't use both values as fixed

File: app/scss/w-input.scss [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+.w-input-button {
+  background: $green;
+  bottom: 0;
+  color: $white;
+  padding: 0 10px;
+  position: absolute;
+  right: 0;
+  top: 0;
+
+  .fa {
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: what is this fa?

File: app/scss/w-input.scss [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  font-style: italic;
+}
+
+::placeholder {
+  color: $dusty-gray;
+  font-style: italic;
+}
+
+.w-select {
+  display: inline-block;
```
Comment (FrankIglesias) [inline app/scss/w-input.scss:39]: avoid copy pasting from the sketch like this. Use px and avoid mixing em and px in a calc

File: app/scss/w-input.scss [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
+  box-sizing: border-box;
+  -webkit-appearance: none;
+  -moz-appearance: none;
+}
+
+.w-select:-moz-focusring {
+  color: transparent;
+  text-shadow: 0 0 0 #000;
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: use color variables

File: app/scss/w-input.scss [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+.w-select {
+  display: inline-block;
+  padding-right: 3.5em;
+  background-image: linear-gradient(50deg, transparent 50%, gray 50%), linear-gradient(125deg, gray 50%, transparent 50%);
+  background-position: calc(100% - 22px) calc(1em + 2px), calc(100% - 12px) calc(1em + 2px), calc(100% - 2.5em) 0.5em;
+  background-size: 7px 10px, 10px 10px, 1px 2em;
+  background-repeat: no-repeat;
+
+  /* reset */
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: remove this comment

File: app/scss/w-input.scss [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  margin: 0;
+  -webkit-box-sizing: border-box;
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: remove this two lines

File: app/scss/w-input.scss [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  -moz-box-sizing: border-box;
+  box-sizing: border-box;
+  -webkit-appearance: none;
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: same with theese

File: app/scss/w-input.scss [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  -moz-appearance: none;
+}
+
+.w-select:-moz-focusring {
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: same

File: app/scss/equalizer.scss [hunk 11]
```diff
@@ -0,0 +1,92 @@
+$equalizer-black: #000;
```
Comment (FrankIglesias) [inline app/scss/equalizer.scss:1]: you can now use [wolox equalizer](https://www.npmjs.com/package/wolox-equalizer) from npm

File: package.json [hunk 12]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "wbooks",
+  "version": "1.0.0",
+  "description": "Wolox Books Training App",
+  "main": "index.js",
+  "scripts": {
+    "postinstall": "npm run copy-libs",
```
Comment (FrankIglesias) [inline package.json:7]: remove some extra tasks here...

File: app/index.html [hunk 13]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<!--[if lt IE 7]>      <html lang="en" ng-app="wBooksApp" class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
```
Comment (FrankIglesias) [inline app/index.html:2]: remove IE lines

File: app/scss/w-input.scss [hunk 14]
```diff
@@ -0,0 +1,58 @@
+@import '_colors';
+
+.w-input-container {
+  display: inline-flex;
```
Comment (FrankIglesias) [inline app/scss/w-input.scss]: flex?

File: app/index.html [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    <li><a href="#!/view2">view2</a></li>
+  </ul> -->
+
+  <!--[if lt IE 7]>
+      <p class="browsehappy">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> to improve your experience.</p>
+  <![endif]-->
+  <div ng-view></div>
+
+
+  <!-- In production use:
```
Comment (FrankIglesias) [inline app/index.html:26]: same

File: app/index.html [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+<head>
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <title>WBooks</title>
+  <meta name="description" content="">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="application.css">
+</head>
+<body>
+  <!-- <ul class="menu">
```
Comment (FrankIglesias) [inline app/index.html:15]: same

**Review by FrankIglesias — CHANGES_REQUESTED** [review]
You can optionally add a linter checker like [this one](https://eslint.org/)

File: app/books/books.component.js [hunk 17]
```diff
@@ -0,0 +1,108 @@
+'use strict';
+
+angular.
+  module('booksList').
+  component('booksList', {
+    templateUrl: 'books/books.template.html',
+    controller: function BooksListController() {
+      var ctrl = this;
+      ctrl.books = [
```
Comment (FrankIglesias) [inline app/books/books.component.js:9]: Think about moving this array to a json file

File: app/scss/equalizer.scss (hunk 11 above)
Comment (eledwinn) [reply app/scss/equalizer.scss:1]: This is in the branch "books-gulp"

File: package.json (hunk 12 above)
Comment (eledwinn) [reply package.json:7]: This is in the branch "books-gulp"

File: app/index.html (hunk 13 above)
Comment (eledwinn) [reply app/index.html:2]: This is in the "books-pug" branch

File: app/index.html (hunk 15 above)
Comment (eledwinn) [reply app/index.html:26]: This is in the "books-pug" branch

File: app/index.html (hunk 16 above)
Comment (eledwinn) [reply app/index.html:15]: This is in the "books-pug" branch

File: app/books/books.component.js (hunk 17 above)
Comment (eledwinn) [reply app/books/books.component.js:9]: This is in the "books-gulp" branch

**Review by FrankIglesias — APPROVED** [review]

**Pull request merged**